### PR TITLE
Add konnectivity-server to controlplane

### PR DESCRIPTION
**Pull request by sjenning**
This is the first in a series of ~4 PRs to add konnectivity support.

The way I see this happening is
* add konnectivity-server `Deployment` to controlplane (this PR)
* add konnectivity-agent `DaemonSet` to user manifests for deployment in guest cluster
* change KAS config to use konnectivity, which will need to include a solution for controlplane side `APIServices`
* remove openvpn

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/kas/egresscfg.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	kasScheme     = runtime.NewScheme()
+	kasSerializer = json.NewSerializerWithOptions(
+		json.DefaultMetaFactory, kasScheme, kasScheme,
+		json.SerializerOptions{Yaml: true, Pretty: true, Strict: true},
+	)
+)
+
+func init() {
+	kasv1beta1.AddToScheme(kasScheme)
+}
```
Comment (ironcladlou) [inline control-plane-operator/controllers/hostedcontrolplane/kas/egresscfg.go:62]: We have a place for this stuff you can use: https://github.com/openshift/hypershift/blob/main/api/scheme.go#L36

**Review by csrwng — COMMENTED** [review]
@sjenning in general this looks good. However, I've been in the process of making the reconcile functions stateless (ie. not members of KonnectivityServerParams) and using the params struct as something that's only helpful in translating hcp + eventual global config into individual pieces of config used by each reconcile function. (See https://github.com/openshift/hypershift/blob/main/control-plane-operator/controllers/hostedcontrolplane/etcd/reconcile.go). It'd be good if you could follow that pattern. The reason is we want the code to explicitly show in code which things are used to reconcile an individual resource.

The other thing I noted is that I see funcs for a KonnectivityServerService that are not called, but the actual service that is reconciled is a KonnectivityAgentService. I'm not sure that the name KonnectivityAgentService makes sense. Please rename KonnectivityAgentService to KonnectivityServerService and remove the duplicate funcs.

**Pull request merged**